includes/functions.php

```diff
@@ -3219,42 +3219,109 @@ function dokan_remove_action_button( $actions ) {
+ * @return string|array
  */
-function dokan_get_translated_days( $day ) {
-    switch ( $day ) {
+function dokan_get_translated_days( $days = true ) {
+    $all_days = [
+        'sunday'    => __( 'Sunday', 'dokan-lite' ),
+        'monday'    => __( 'Monday', 'dokan-lite' ),
+        'tuesday'   => __( 'Tuesday', 'dokan-lite' ),
+        'wednesday' => __( 'Wednesday', 'dokan-lite' ),
+        'thursday'  => __( 'Thursday', 'dokan-lite' ),
+        'friday'    => __( 'Friday', 'dokan-lite' ),
+        'saturday'  => __( 'Saturday', 'dokan-lite' ),
+    ];
+
+    // Get days array if our $days is true.
+    if ( true === $days ) {
+        return $all_days;
+    }
+
+    // Translated only day if passed the name of current day.
+    switch ( $days ) {
         case 'saturday':
-            return __( 'Saturday', 'dokan-lite' );
+            return $all_days['saturday'];
 
         case 'sunday':
-            return __( 'Sunday', 'dokan-lite' );
+            return $all_days['sunday'];
 
         case 'monday':
-            return __( 'Monday', 'dokan-lite' );
+            return $all_days['monday'];
 
         case 'tuesday':
-            return __( 'Tuesday', 'dokan-lite' );
+            return $all_days['tuesday'];
 
         case 'wednesday':
-            return __( 'Wednesday', 'dokan-lite' );
+            return $all_days['wednesday'];
 
         case 'thursday':
-            return __( 'Thursday', 'dokan-lite' );
+            return $all_days['thursday'];
 
         case 'friday':
-            return __( 'Friday', 'dokan-lite' );
+            return $all_days['friday'];
 
         case 'close':
             return apply_filters( 'dokan_store_close_day_label', __( 'Off Day', 'dokan-lite' ) );
 
         default:
-            return apply_filters( 'dokan_get_translated_days', '', $day );
-            break;
+            return apply_filters( 'dokan_get_translated_days', '', $days );
+    }
+}
+
+/**
+ * Get store opening time.
+ *
+ * @since 3.2.16
+ *
+ * @param string $current_day
+ * @param int    $index
+ *
+ * @return mixed|string
+ */
+function dokan_get_store_opening_time( $current_day, $index = 0 ) {
+    $profile_info       = dokan_get_store_info( dokan_get_current_user_id() );
+    $dokan_store_time   = isset( $profile_info['dokan_store_time'] ) ? $profile_info['dokan_store_time'] : '';
+    $dokan_opening_time = isset( $dokan_store_time[ $current_day ]['opening_time'] ) ? $dokan_store_time[ $current_day ]['opening_time'] : '';
+
+    if ( ! empty( $dokan_opening_time ) && ! is_array( $dokan_opening_time ) ) {
+        return $dokan_opening_time;
+    }
+
+    if ( ! empty( $dokan_opening_time ) && is_array( $dokan_opening_time ) ) {
+        return $dokan_opening_time[ $index ];
     }
+
+    return '';
+}
+
+/**
+ * Get store closing time.
+ *
+ * @since 3.2.16
+ *
+ * @param string $current_day
+ * @param int    $index
+ *
+ * @return mixed|string
+ */
+function dokan_get_store_closing_time( $current_day, $index = 0 ) {
```

Comment:
These two functions implement almost same type of things. We should follow the `DRY concept`. Better would be, create just one function. And we could also generate two function from these in future.
```php

/*
* @params string $time_type, eg: `opening_time` or `closing_time`
*/
function dokan_get_store_time( $time_type, $current_day ) { // You don't use the index anywhere also, so why's that?
     
  // just some snippet would be like this - 
  $store_time = isset( $store_info['dokan_store_time'] ) ? $store_info['dokan_store_time'] : '';
  $time       = isset( $store_time[ $current_day ][$time_type] ) ? $store_time[ $current_day ][$time_type] : '';
 // do rest as this
}
```

Then in future use case - 
```php
dokan_get_store_time( 'opening_time', 1 );
dokan_get_store_time( 'closing_time', 1 );
```